workspace/resource_workspace_file.go

```diff
@@ -37,21 +38,28 @@ func ResourceWorkspaceFile() *schema.Resource {
 				return err
 			}
 			path := d.Get("path").(string)
-			parent := filepath.ToSlash(filepath.Dir(path))
-			if parent != "/" {
-				err = client.Workspace.MkdirsByPath(ctx, parent)
-				if err != nil {
-					return err
-				}
-			}
-			err = client.Workspace.Import(ctx, ws_api.Import{
+			importReq := ws_api.Import{
 				Content:   base64.StdEncoding.EncodeToString(content),
 				Format:    ws_api.ImportFormatAuto,
 				Path:      path,
 				Overwrite: true,
-			})
+			}
+			err = client.Workspace.Import(ctx, importReq)
 			if err != nil {
-				return err
+				if isParentDoesntExistError(err) {
+					parent := filepath.ToSlash(filepath.Dir(path))
+					if parent != "/" {
```

Comment:
Ditto, we probably can inline the if body here since it will always be true

Reply:
No - this is to handle of cases if I notebooks/files are created at root level